swarm/netsim/http.go

```diff
@@ -17,45 +17,34 @@
 package netsim
 
 import (
-	"context"
 	"fmt"
 	"net/http"
 
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/p2p/simulations"
 )
 
-func (s *Simulation) initHTTPServer(opts *SimulationOptions) {
-	//assign default port if nothing provided
-	if opts.HTTPSimPort == "" {
-		opts.HTTPSimPort = DefaultHTTPSimPort
+// Package defaults.
+var (
+	DefaultHTTPSimAddr = ":8888"
+)
+
+func (s *Simulation) startHTTPServer(addr string) {
```

Comment:
Can this method be called WithServer and to remove the method with the same name in simulation.go https://github.com/ethersphere/go-ethereum/pull/773/files#diff-729494f5385639fe02fb100e48e49382R102 that just wraps startHTTPServer?